runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/middleware/AwsSigningMiddleware.kt

```diff
@@ -190,3 +212,8 @@ private fun HttpRequestBuilder.update(signedRequest: HttpRequest) {
         }
     }
 }
+
+/**
+ * @return a boolean representing if this HttpBody is large enough to send via aws-chunked
+ */
+private fun HttpBody.isEligibleForAwsChunkedStreaming(): Boolean = (this.contentLength ?: 0) > AwsSigningMiddleware.AWS_CHUNKED_THRESHOLD
```

Comment:
**Style**: This seems like a natural use case for extension `val` (rather than extension `fun`).